### PR TITLE
Clean docs

### PR DESCRIPTION
**Pull request by Hackdom**
Good documentation and clean code brings bugs from under the rug

### CODE REVIEW

File: Wallet.sol [hunk 1]
```diff
@@ -24,36 +34,31 @@ contract multiowned {
...
-	// some others are in the case of an owner changing.
-	event OwnerChanged(address oldOwner, address newOwner);
-	event OwnerAdded(address newOwner);
-	event OwnerRemoved(address oldOwner);
-	// the last one is emitted if the required signatures change
-	event RequirementChanged(uint newRequirement);
+	event Confirmation(address owner, bytes32 operation); //record confirmation with owner and hash
+	event Revoke(address owner, bytes32 operation); //record revocation with owner and hash
+	event OwnerChanged(address oldOwner, address newOwner); //owner change
+	event OwnerAdded(address newOwner); //self expalantory
```
Comment (5chdn) [inline Wallet.sol]: typo: _explanatory_, also, do you really need a comment for this? I'd prefer to just write `owner addition`, even though it's self-explanatory

File: Wallet.sol [hunk 2]
```diff
@@ -149,29 +172,28 @@ contract multiowned {
...
-		// determine the bit to set for this owner.
-		uint ownerIndexBit = 2**ownerIndex;
+
+		uint ownerIndexBit = 2**ownerIndex; // determine the bit to set for this owner.
 		// make sure we (the message sender) haven't confirmed this operation previously.
 		if (pending.ownersDone & ownerIndexBit == 0) {
 			Confirmation(msg.sender, _operation);
 			// ok - check if count is enough to go ahead.
-			if (pending.yetNeeded <= 1) {
+			if (pending.yetNeeded == 1) {
```
Comment (5chdn) [inline Wallet.sol]: This is fine, however, just leaving a comment to highlight it's not just docs.

**Comment by Hackdom** [comment]
I was pulling apart a previously overloaded commit, 100% my fault on that one. That was to make the logic more stringent, looks like the the logic is negligible but the standard should be more strict is what I was getting at there.

**Comment by gavofyork** [comment]
prefer to keep it strictly docs as a matter of process.

**Comment by Hackdom** [comment]
Updated and removed the logic, changed wording in comment

**Comment by Hackdom** [comment]
I don't see that doCreate function in the contract and so this pulled out the modified create() function

**Comment by gavofyork** [comment]
i split the `create` out since later versions of solidity complain about name shadowing between the function and the asm operation.

**Comment by Hackdom** [comment]
Retuned modified create() function

**Comment by Ivshti** [comment]
@gavofyork ok, this

> i split the create out since later versions of solidity complain about name shadowing between the function and the asm operation.

answers one of the things in the audit

Other than that, I'm a bit worried about changing whatever else other than documentation. Other than that, I need to say, the code DOES need a lot of comments.

**Comment by gavofyork** [comment]
> I'm a bit worried about changing whatever else other than documentation

i agree. would be best to split this further into pure documentation alterations, with the code cut'n'pasting be separate.

**Comment by Hackdom** [comment]
I'm back and changing it up.. Saw the gitter channel also

**Comment by Hackdom** [comment]
Ok, I changed just docs. I'll submit one more pull to move the fields up because it reads better when you have the contract variables up front and then the methods after. Also, the create function in the wallet is not under the Internal Methods label when it is in fact an internal method.

File: Wallet.sol [hunk 3]
```diff
@@ -42,18 +42,18 @@ contract multiowned {
 		if (isOwner(msg.sender))
 			_;
 	}
-	// multi-sig function modifier: the operation must have an intrinsic hash in order
-	// that later attempts can be realised as the same underlying operation and
-	// thus count as confirmations.
+
+   //_operation is a hash to confirm transaction with others using same hash
```
Comment (gavofyork) [inline Wallet.sol]: this is hardly better documentation. the original comment explained more and had proper punctuation to boot.

**Comment by gavofyork** [comment]
ok mostly good now; aside from the previous comment, it would be nice to keep the whitespace consistent - this file is tabs only.

File: Wallet.sol (hunk 3 above)
Comment (Hackdom) [reply Wallet.sol]: It was just to shorten but I see your point.

**Comment by Hackdom** [comment]
ok, let me pull this again and fix the tabs. Two spaces right?

**Comment by Hackdom** [comment]
ha, ok... need a fresh start. git's all backwards.

**Pull request closed without merging**